bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/votable/VoTableClient.scala

```diff
@@ -39,21 +42,31 @@ trait VoTableClient {
     p.future
   }
 
-  protected def doQuery(query: CatalogQuery, url: String): Future[QueryResult] = future {
-    val method = new GetMethod(s"$url/cgi-bin/conesearch.py")
-    val qs = queryParams(query)
-    method.setQueryString(qs)
+  // weak hash map should clean itself if we are using too much memory
+  val memo:Memo[(CatalogQuery, String), QueryResult] = Memo.weakHashMapMemo
```

Comment:
Since two different `(CatalogQuery, String)` pairs could hash to the same `Int` you wouldn't know whether the value really corresponds.  Seems like they should provide an LRU cache implementation, which is easily obtainable via `LinkedHashMap`.